draft/affordances.md

```diff
@@ -0,0 +1,56 @@
+---
+RFC: XXXX
+Author: Kyle Fuller
+Status: Draft
+Created: 2015-09-21
+Last Modified: 2015-09-21
+---
+
+# API Blueprint RFC XXXX: Affordances
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Rationale](#rationale)
+    - [Example](#example)
+
+## Abstract
+
+This RFC proposes that you can define affordances (link relations) that an API
+resource or action provides.
+
+## Rationale
+
+This RFC proposes adding an `Affordances` section into API Blueprint that
+sits under a resource or action providing the affodances that it provides.
+
+### Example
+
+To describe affordances from a `Question` resource to two other resources, we
+can describe them within the affordances section as follows:
+
+```apib
+## Question [/question/{id}]
+
++ Parameters
+    + id: 1
+
++ Affordances
+    + `create_comment` (Comments[create])
+    + `author` (User)
```

Comment:
I think I'm missing something because it's not clear to me at all what this is solving, either by the example or the description above. I understand that the goal is to allow a client to understand link relationships between resources or arbitrary actions, but what does the above actually get me as a client? I'm wondering:

* What are the `create_comment` and `author` strings? Are they just an identifier or would an implementor be expected to use these to provide something like e.g. `my_question.create_comment(params)`? I think any update to the spec should describe why these names exist.

* How would I access the author? Presumably the question would have an attribute called `author_id` or something that would be taken as a parameter into the `author` relationship when calling `/user/{username}`. Where is the link between `my_question.author_id` and `/user/{username}` defined here? What can I actually do without that link? I know an author may exist and that the author may have a URL, but really nothing else.

* Similar to the previous point, I assume `create_comment` would require some kind of `question_id` attribute and you'd want to link `my_question.id` to it, but how with the information provided?

* In the `user` example, how do you link it to any specific action on the user? How do I know whether to use GET or PUT or something else?